subprojects/platform-play/src/integTest/groovy/org/gradle/play/integtest/continuous/PlayReloadIntegrationTest.groovy

```diff
@@ -196,11 +204,130 @@ task otherTask {
         appIsRunningAndDeployed()
 
         when:
-        addHelloWorld()
+        addNewRoute()
 
         then:
         fails()
         !executedTasks.contains('runPlayBinary')
         errorPageHasTaskFailure("otherTask")
     }
+
+    void addPendingChangesHook() {
+        buildFile << """
+            ext.pendingChanges = new java.util.concurrent.atomic.AtomicBoolean(false)
+
+            void addPendingChangeListener() {
+                def pendingChangesManager = gradle.services.get(${PendingChangesManager.canonicalName})
+                pendingChangesManager.addListener new ${SingleFirePendingChangesListener.canonicalName}({
+                    synchronized(pendingChanges) {
+                        println "Pending changes detected"
+                        pendingChanges.set(true)
+                        pendingChanges.notifyAll()
+                    }
+                } as ${PendingChangesListener.canonicalName})
+            }
+
+            void waitForPendingChanges() {
+                synchronized(pendingChanges) {
+                    while(!pendingChanges.get()) {
+                        pendingChanges.wait()
+                    }
+                }
+            }
+        """
+    }
+
+    def "wait for changes to be built when a request comes in during initial app startup and there are pending changes"() {
```

Comment:
This should be similar to the other test cases, except that the blocking task would be enabled right from the start of the build, rather than after the initial build succeeds and the app is running.